config/areas/site/views.php

```diff
@@ -14,6 +14,28 @@
 			return Find::file('pages/' . $id, $filename)->panel()->view();
 		}
 	],
+	'page.preview' => [
+		'pattern' => 'pages/(:any)/preview/(changes|latest|compare)',
+		'action'  => function (string $path, string $mode) {
+			$page    = Find::page($path);
+			$view    = $page->panel()->view();
+			$preview = $page->previewUrl();
+
+			return [
+				'component' => 'k-preview-view',
+				'props'     => [
+					...$view['props'],
+					'back' => $view['props']['link'],
+					'mode' => $mode,
+					'src'  => [
+						'changes' => $preview . '?_version=changes',
```

Comment:
`Todo:` Doesn't work for draft URLs that already include the `?token=` query. See general discussion.

Reply:
I've fixed it by adding a new VersionId option to the `Site::previewUrl` and `Page::previewUrl` methods.